### DISCUSSION TITLE
Hide underlying exe installer from Add/Remove programs

### DISCUSSION THREAD

**Discussion by LSPieterS** [wixtoolset/issues#8855] [Questions]
Hi,

I've been trying to wrap an existing exe installer with wix toolset so that it's possible to install/uninstall/upgrade using an MSI. What I've noticed is that the installation creates 2 entries in Add/Remove programs, one for the MSI and one for the underlying exe installer. The goal is to only see the MSI entry in that list. Is there an option in wix toolset to hide the one from the exe installer? I have tried using a custom action with a DLL to add the SystemComponent key and with a PowerShell script doing the same but none of this seemed to do the trick.

Any suggestions or useful documentation?

Thanks in advance!
Pieter

**Comment by wmanning** [comment]
That's not possible unless the EXE has a command line argument that tell it not to add a ARP (add/remove programs) entry.
It is possible with MSI packages though.

**Reply by LSPieterS** [reply]
Indeed, it's possible to generate the EXE installer so it doesn't add the ARP entry, but then the MSI uninstallation fails.

**Comment by wmanning** [comment]
I thought you were using burn for some reason. You are using an MSI to also run an (installer) exe.  That's usually not recommended as there should be only one installer running at a time (modifying machine state).  

Do you know what install technology the exe intaller users (Inno, MSI wrapped in an EXE, ...)?

In [burn](https://wixtoolset.org/docs/tools/burn/) you could use `<MsiPackage>` and `<ExePackage>` to install your MSI and EXE installer.  Burn is like a bootstrapper, it creates an executable that installs several packages.  But if you need a MSI only then I don't see how. Burn also is meant to replace the UI of the MSI and that may be an issue if you have a lot invested in a MSI UI.

**Reply by wmanning** [reply]
Just to clarify the wixtoolset can be used to create two main types of installers:
1. MSI
2. EXE (burn) - which can install one or more MSI, other burn EXEs, and generic installer EXEs (say from Inno or ).

**Reply by wmanning** [reply]
> Indeed, it's possible to generate the EXE installer so it doesn't add the ARP entry, but then the MSI uninstallation fails.

Not sure what is happening there.
